src/App.tsx

```diff
@@ -1,34 +1,93 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [isLoadingTodos, setIsLoadingTodos] = useState(false);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setIsLoadingTodos(true);
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setFilteredTodos(todosFromServer);
+      })
+      .finally(() => setIsLoadingTodos(false));
+  }, []);
+
+  useEffect(() => {
+    let filtered = todos;
+
+    if (filterStatus !== 'all') {
+      filtered = todos.filter(todo =>
+        filterStatus === 'completed' ? todo.completed : !todo.completed,
+      );
+    }
+
+    if (query) {
+      filtered = filtered.filter(todo =>
+        todo.title.toLowerCase().includes(query.toLowerCase()),
+      );
+    }
+
+    setFilteredTodos(filtered);
+  }, [todos, filterStatus, query]);
+
+  const handleSelectTodo = (todoId: number | null) => {
+    setSelectedTodoId(todoId);
+  };
+
+  const closeModal = () => {
+    setSelectedTodoId(null);
+  };
```

Comment:
The naming convention for the method is not followed. It should be 'onCloseModal' instead of 'closeModal'. Please refer to the checklist link for naming conventions.